streaming/src/main/scala/sparkAtScale/StreamingDirectEmails.scala

```diff
@@ -79,6 +79,21 @@ object StreamingDirectEmails {
       //println(s"ssc: $ssc")
       println(s"kafkaParams: $kafkaParams")
 
+      // Update the cumulative count of emails received using mapWithState
+      // This will give a DStream made of state (which is the cumulative count of the keys)
+      val mappingFunc = (email_key: String, email_body: Option[String], state: State[Long]) => {
+          var newMessageCount:Option[Long] = None 
+
+          // Check if state exists
+          if (state.exists) {
+              newMessageCount = Some(state.get + 1L)
+          } else {
+              newMessageCount = Some(0L) 
+          }
```

Comment:
Minor nit but for code like this I always push the conditional further down (allowing it to be a value): `val newMessageCount = Some(if state.exists (state.get + 1L) else 0L)`